packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -312,6 +312,25 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
         ? itemsWithoutPositions.slice(splitIndex, splitIndex + TWO_COL_ITEMS_MEASURE_BATCH_SIZE)
         : itemsWithoutPositions;
 
+      if (isFirstRow) {
+        const remainingColumnsToFill = columnCount - pre.length; // different of column count minus number of columns _before_ two column module position
+        const multiColumnItemWidth = parseInt(twoColumnItems[0].columnSpan, 10);
+        if (multiColumnItemWidth > remainingColumnsToFill) {
+          // if we're in the first row but we can't render the multi column item in the first row, fill pre with single column items to ensure we have enough to cover the initial row
+          // this is specifically for the first row to ensure we match server rendering styling
+          for (let i = 0; i < remainingColumnsToFill; i += 1) {
+            const singleColumnItemIndex = batchWithTwoColumnItems.findIndex(
+              (item) => !item.columnSpan || item.columnSpan === 1,
+            );
+            if (singleColumnItemIndex > -1) {
+              // if we find a single column module, remove it from the two column batch and add it to the pre array
+              pre.push(batchWithTwoColumnItems[singleColumnItemIndex]);
+              batchWithTwoColumnItems.splice(singleColumnItemIndex, 1);
```

Comment:
This is effectively reducing the number of elements on `batchWithTwoColumnItems` so the two column module will have less elements to calculate the optimal position.

Reply:
ah good callout. let me update